### PR TITLE
[Merged by Bors] - feat: some lemmas about associated and prime elements

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
From flt-regular.

Co-authored-by: Andrew Yang <the.erd.one@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by timotree3** [comment]
Pushed a commit hopefully fixing CI. One of the new lemmas was showing up in the output of a `have?` test.

**Comment by alexjbest** [comment]
LGTM

maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/Algebra/Associated.lean [hunk 1]
```diff
@@ -60,6 +60,12 @@ theorem dvd_of_dvd_pow (hp : Prime p) {a : α} {n : ℕ} (h : p ∣ a ^ n) : p 
   exact ih dvd_pow
 #align prime.dvd_of_dvd_pow Prime.dvd_of_dvd_pow
 
+theorem dvd_mul {a b : α} : p ∣ a * b ↔ p ∣ a ∨ p ∣ b :=
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Associated.lean:63]: Should the name contain `iff`? (Not sure.)

Comment (Ruben-VandeVelde) [reply Mathlib/Algebra/Associated.lean:63]: Only if we also change `Nat.Prime.dvd_mul`

**Pull request closed without merging**